tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/NewPcmUmlClassApplicationTest.xtend

```diff
@@ -0,0 +1,140 @@
+import org.palladiosimulator.pcm.repository.RepositoryFactory
+import tools.vitruv.applications.pcmumlclass.CombinedPcmToUmlClassReactionsChangePropagationSpecification
+import tools.vitruv.applications.pcmumlclass.CombinedUmlClassToPcmReactionsChangePropagationSpecification
+import tools.vitruv.framework.views.View
+import tools.vitruv.testutils.ViewBasedVitruvApplicationTest
+import tools.vitruv.applications.pcmumlclass.DefaultLiterals
+import static org.hamcrest.MatcherAssert.assertThat
+import static tools.vitruv.testutils.matchers.ModelMatchers.isNoResource
+import static tools.vitruv.testutils.matchers.ModelMatchers.isResource
+
+import static extension tools.vitruv.applications.pcmumlclass.tests.PcmQueryUtil.*
+import static extension tools.vitruv.applications.pcmumlclass.tests.UmlQueryUtil.*
+
+abstract class NewPcmUmlClassApplicationTest extends ViewBasedVitruvApplicationTest {
+
+	@Accessors(PROTECTED_GETTER)
+	static val MODEL_FILE_EXTENSION = "uml"
+	@Accessors(PROTECTED_GETTER)
+	static val MODEL_NAME = "model"
+	@Accessors(PROTECTED_GETTER)
+	static val MODEL_FOLDER_NAME = "model"
+
+	static val PACKAGE_NAME = "rootpackage"
+	static val PACKAGE_NAME_FIRST_UPPER = "Rootpackage"
+	static val CONTRACTS_PACKAGE = "contracts"
+
+	protected var extension PcmUmlclassViewFactory viewFactory
+	protected var PcmUmlClassApplicationTestHelper helper
+	protected var ResourceSet testResourceSet
+
+	override protected getChangePropagationSpecifications() {
+		return #[
+			new CombinedPcmToUmlClassReactionsChangePropagationSpecification,
+			new CombinedUmlClassToPcmReactionsChangePropagationSpecification
+		]
+	}
+
+	protected def getTestResource(URI uri) {
+		return testResourceSet.getResource(uri, true)
+	}
+
+	protected def void createUmlModel((Model)=>void modelInitialization) {
+		changeUmlView [
+			val umlModel = UMLFactory.eINSTANCE.createModel
+			createAndRegisterRoot(umlModel, MODEL_NAME.projectModelPath.uri)
+			modelInitialization.apply(umlModel)
+		]
+	}
+
+	protected def void createPcmRepository(String packageName) {
+		changePcmView [
+			val repository = RepositoryFactory.eINSTANCE.createRepository => [
+				it.entityName = packageName
+			]
+			createAndRegisterRoot(repository, MODEL_NAME.projectModelPath.uri)
+		]
+	}
+
+	protected def void createAndRegisterRoot(View view, EObject rootObject, URI persistenceUri) {
+		view.registerRoot(rootObject, persistenceUri)
+	}
+
+	protected def Path getProjectModelPath(String modelName) {
+		Path.of(MODEL_FOLDER_NAME).resolve(modelName + "." + MODEL_FILE_EXTENSION)
+	}
+
+	protected def getDefaultUmlModel(View view) {
+		view.claimUmlModel(MODEL_NAME)
+	}
+
+	protected def getDefaultPcmRepository(View view) {
+		view.claimPcmRepository(PACKAGE_NAME_FIRST_UPPER)
+	}
+
+	def void init_PCM() {
+		userInteraction.addNextTextInput(PcmUmlClassApplicationTestHelper.UML_MODEL_FILE)
+		createPcmRepository(PACKAGE_NAME_FIRST_UPPER)
+	}
+
+	def void init_UML() {
```

Comment:
Are the underscores in the method names on purpose?

Reply:
Renamed methods in [e6b7cad](https://github.com/vitruv-tools/Vitruv-CaseStudies/pull/214/commits/e6b7cad244960ae56d1701157e18513ae1d0ea9e)